src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -63,17 +67,28 @@ internal class TestRequestManager : ITestRequestManager
             TestPlatformFactory.GetTestPlatform(),
             TestLoggerManager.Instance,
             TestRunResultAggregator.Instance,
-            TestPlatformEventSource.Instance)
+            TestPlatformEventSource.Instance,
+            IsTelemetryOptedOut() ? (IMetricsPublisher)new NoOpMetricsPublisher() : new MetricsPublisher())
         {
         }
 
-        internal TestRequestManager(CommandLineOptions commandLineOptions, ITestPlatform testPlatform, TestLoggerManager testLoggerManager, TestRunResultAggregator testRunResultAggregator, ITestPlatformEventSource testPlatformEventSource)
+        /// <summary>
+        /// Finalizes an instance of the <see cref="TestRequestManager"/> class. 
+        /// </summary>
+        ~TestRequestManager()
+        {
+            this.metricsPublisher.Dispose();
+        }
```

Comment:
nit: Can you please move this to the end.

Reply:
removed